jobqueue/jobqueue.go

```diff
@@ -91,34 +91,36 @@ func (q *jobQueue) Pop(limit uint) ([]Job, error) {
 }
 
 func (q *jobQueue) Complete(job Job, res *Result) {
-	j := &completedJob{job, 0}
-	if res.IsFailure() {
-		q.stats.fail(1)
-		j.failed = 1
+	var j *completedJob
+	if res.Status == ResultStatusSuccess {
```

Comment:
Should we abstract conditions like this with methods like `res.IsSuccess()`?

Reply:
Yes, that would be nice.